File: src/main/java/org/sonar/plugins/xquery/checks/AbstractProhibitLibraryCheck.java
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.checks;

import org.sonar.plugins.xquery.parser.XQueryParser;
import org.sonar.plugins.xquery.parser.XQueryTree;

public abstract class AbstractProhibitLibraryCheck extends AbstractCheck {

    @Override
    public void enterExpression(XQueryTree node) {
        super.enterExpression(node);

        if (XQueryParser.ModuleNamespace == node.getType()) {
            String namespace = node.getValue("StringLiteral");
            if (getNamespace().equals(namespace)) {
                createIssue(node.getLine());
            }
        }
    }

    abstract protected void createIssue(int lineNumber);
    abstract protected String getNamespace();
}

File: src/main/java/org/sonar/plugins/xquery/checks/AbstractProhibitLibraryCheck.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.checks

import org.antlr.v4.runtime.ParserRuleContext
import org.sonar.plugins.xquery.parser.XQueryParser.ModuleDeclContext
import org.sonar.plugins.xquery.parser.getLine

abstract class AbstractProhibitLibraryCheck : AbstractCheck() {
    protected abstract val namespace: String

    override fun enterExpression(node: ParserRuleContext) {
        super.enterExpression(node)

        if (node is ModuleDeclContext) {
            val namespace = node.uri.text
            if (namespace == namespace) {
                createIssue(node.getLine())
            }
        }
    }

    protected abstract fun createIssue(lineNumber: Int)
}
